src/pages/index.js

```diff
@@ -63,37 +112,106 @@ function openEditProfileModal() {
   editProfileModal.open();
 }
 
-const newCardModal = new PopupWithForm({
-  popupSelector: "#add-card-modal",
-  handleFormSubmit: submitCard,
-});
+function handleProfileEditSubmit({ title, description }) {
+  editProfileModal.setLoading(true);
+  api
+    .editUserInfo(title, description)
+    .then(() => {
+      userInfo.setUserInfo(title, description);
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editProfileModal.setLoading(false, "Save");
+    });
+}
 
-const imageModal = new PopupWithImage({
-  popupSelector: "#preview-image-modal",
-});
+function handleProfileAvatarSubmit({ url }) {
+  avatarEditModal.setLoading(true);
+  api
+    .editUserAvatar(url)
+    .then((userData) => {
+      userInfo.setProfileAvatar(userData.avatar);
+      avatarEditModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarEditModal.setLoading(false, "Save");
+    });
+}
+
+/*Card Edit*/
+function handleCardSubmit({ title, url }) {
+  newCardModal.setLoading(true);
+  api
+    .addNewCard(title, url)
+    .then((card) => {
+      const newCard = createCard(card);
+      cardListSection.prependItem(newCard);
+      newCardModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      newCardModal.setLoading(false, "Create");
+    });
+}
 
-/*Functions*/
-function createCard({ name, link }) {
-  const cardElement = new Card(
-    { name, link },
+function createCard(data) {
+  const newCard = new Card(
+    data,
+    userId,
     "#card-template",
-    ({ name, link }) => {
-      imageModal.open({ name, link });
+    function handleCardClick(data) {
+      imagePreviewModal.open(data);
+    },
+    function handleDeleteClick() {
+      cardDeleteModal.setSubmitAction(() => {
+        cardDeleteModal.setLoading(true);
+        api
+          .deleteCard(data._id)
+          .then((res) => {
+            newCard.remove(res._id);
+            cardDeleteModal.close();
+          })
+          .catch((err) => {
+            console.error(err);
+          })
+          .finally(() => {
+            cardDeleteModal.setLoading(false, "Yes");
+          });
+      });
+      cardDeleteModal.open(data._id);
+    },
+    function handlelikeClick(data) {
+      api
+        .likeCard(data._id, newCard.isLiked())
+        .then((res) => {
+          const likes = res.likes || [];
+          newCard.setLikes(likes);
+          newCard._renderLikes();
```

Comment:
Private methods should not be used outside of a class. Please call this method inside `setLikes` method